exonum-java-binding-core/src/test/java/com/exonum/binding/storage/indices/ProofListContainsMatcher.java

```diff
@@ -35,10 +33,11 @@
 
 class ProofListContainsMatcher extends TypeSafeMatcher<ProofListIndexProxy<String>> {
 
-  private final Function<ProofListIndexProxy<String>, ListProof> proofFunction;
+  private final Function<ProofListIndexProxy<String>, UncheckedListProof> proofFunction;
   private final Matcher<Map<Long, String>> elementsMatcher;
 
-  private ProofListContainsMatcher(Function<ProofListIndexProxy<String>, ListProof> proofFunction,
+  private ProofListContainsMatcher(Function<ProofListIndexProxy<String>,
```

Comment:
I'd consider such formatting, to avoid confusion:
```java
  private ProofListContainsMatcher(
      Function<ProofListIndexProxy<String>, UncheckedListProof> proofFunction,
      Map<Long, String> expectedProofElements) {
```